src/soundsourceproxy.cpp

```diff
@@ -47,17 +20,24 @@
 #include "sources/soundsourcemodplug.h"
 #endif
 #include "sources/soundsourceflac.h"
+
 #include "util/cmdlineargs.h"
-#include "util/regex.h"
+
+#include <QApplication>
+#include <QDesktopServices>
 
 //Static memory allocation
-QRegExp SoundSourceProxy::m_supportedFileRegex;
-QMap<QString, QLibrary*> SoundSourceProxy::m_plugins;
-QMap<QString, getSoundSourceFunc> SoundSourceProxy::m_extensionsSupportedByPlugins;
-QMutex SoundSourceProxy::m_extensionsMutex;
+Mixxx::SoundSourceProviderRegistry SoundSourceProxy::s_soundSourceProviders;
+
+namespace {
+
+#ifdef __APPLE__
+const QStringList SOUND_SOURCE_PLUGIN_FILENAME_PATTERN("libsoundsource*");
+#else
+// NOTE(uklotzde): Why no filtering for Linux/Windows?
```

Comment:
On Linux the filter should work as well or do I miss something?  
In windows the lib postfix might be missing.